tests/conftest.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+from datetime import datetime
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    outcome = yield
+    report = outcome.get_result()
+    report.start = call.start
+    report.stop = call.stop
+
+def pytest_terminal_summary(terminalreporter):
+    terminalreporter.ensure_newline()
+    terminalreporter.section('start/stop times', sep='-', bold=True)
+    for stat in terminalreporter.stats.values():
+        for report in stat:
+            if report.when == 'call' and report.failed and hasattr(report, 'start') and hasattr(report, 'stop'):
```

Comment:
@italovalcy when I was trying out this PR to see how the terminal report would look like, it has resulted in this traceback, does this error also show up on your branch? Could you double check this? 

```

diff --git a/kytos-init.sh b/kytos-init.sh
index 7fe5c13..e1940f1 100755
--- a/kytos-init.sh
+++ b/kytos-init.sh
@@ -16,7 +16,7 @@ sed -i 's/WARNING/INFO/g' /etc/kytos/logging.ini

 test -z "$TESTS" && TESTS=tests/

-python -m pytest $TESTS 2>&1 | tee log-e2e-$(date +%Y%m%d%H%M%S)
+python -m pytest $TESTS -k test_015_create_evc_intra_switch 2>&1 | tee log-e2e-$(date +%Y%m%d%H%M%S)
```

```
Starting enhanced syslogd: rsyslogd.
/etc/openvswitch/conf.db does not exist ... (warning).
Creating empty database /etc/openvswitch/conf.db.
Starting ovsdb-server.
Configuring Open vSwitch system IDs.
Starting ovs-vswitchd.
Enabling remote OVSDB managers.
+ sed -i 's/STATS_INTERVAL = 60/STATS_INTERVAL = 3/g' /var/lib/kytos/napps/kytos/of_core/settings.py
+ sed -i 's/LINK_UP_TIMER = 10/LINK_UP_TIMER = 1/g' /var/lib/kytos/napps/kytos/topology/settings.py
+ sed -i 's/DEPLOY_EVCS_INTERVAL = 60/DEPLOY_EVCS_INTERVAL = 5/g' /var/lib/kytos/napps/kytos/mef_eline/settings.py
+ sed -i 's/BOX_RESTORE_TIMER = 0.1/BOX_RESTORE_TIMER = 0.5/' /var/lib/kytos/napps/kytos/flow_manager/settings.py
+ sed -i 's/COOKIE_PREFIX = 0xbb/COOKIE_PREFIX = 0xab/g' /var/lib/kytos/napps/kytos/of_lldp/settings.py
+ kytosd --help
+ sed -i s/WARNING/INFO/g /etc/kytos/logging.ini
+ test -z ''
+ TESTS=tests/
+ python -m pytest tests/ -k test_015_create_evc_intra_switch
++ date +%Y%m%d%H%M%S
+ tee log-e2e-20220113131445
============================= test session starts ==============================
platform linux2 -- Python 2.7.16, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
rootdir: /tests, inifile:
plugins: timeout-1.3.3
collected 189 items / 188 deselected

tests/test_e2e_10_mef_eline.py .                                         [100%]

------------------------------- start/stop times -------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/lib/python2.7/dist-packages/pytest.py", line 91, in <module>
    raise SystemExit(pytest.main())
  File "/usr/lib/python2.7/dist-packages/_pytest/config/__init__.py", line 77, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/usr/lib/python2.7/dist-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/lib/python2.7/dist-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/lib/python2.7/dist-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/usr/lib/python2.7/dist-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/usr/lib/python2.7/dist-packages/pluggy/callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "/usr/lib/python2.7/dist-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/usr/lib/python2.7/dist-packages/_pytest/main.py", line 217, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/usr/lib/python2.7/dist-packages/_pytest/main.py", line 210, in wrap_session
    session=session, exitstatus=session.exitstatus
  File "/usr/lib/python2.7/dist-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/lib/python2.7/dist-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/lib/python2.7/dist-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/usr/lib/python2.7/dist-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/usr/lib/python2.7/dist-packages/_pytest/terminal.py", line 641, in pytest_sessionfinish
    terminalreporter=self, exitstatus=exitstatus
  File "/usr/lib/python2.7/dist-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/usr/lib/python2.7/dist-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/lib/python2.7/dist-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/usr/lib/python2.7/dist-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/usr/lib/python2.7/dist-packages/pluggy/callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "/usr/lib/python2.7/dist-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/tests/conftest.py", line 17, in pytest_terminal_summary
    if report.when == 'call' and report.failed and hasattr(report, 'start') and hasattr(report, 'stop'):
AttributeError: 'Function' object has no attribute 'when'
```

Reply:
@viniarck can you please try one more time? I've just added commit bc3c871 to handle this case (in fact the report should only be called for failing tests)

```
# export TESTS="tests/ -k test_015_create_evc_intra_switch"
# ./kytos-init.sh
+ sed -i 's/STATS_INTERVAL = 60/STATS_INTERVAL = 3/g' /var/lib/kytos/napps/kytos/of_core/settings.py
+ sed -i 's/LINK_UP_TIMER = 10/LINK_UP_TIMER = 1/g' /var/lib/kytos/napps/kytos/topology/settings.py
+ sed -i 's/DEPLOY_EVCS_INTERVAL = 60/DEPLOY_EVCS_INTERVAL = 5/g' /var/lib/kytos/napps/kytos/mef_eline/settings.py
+ sed -i 's/BOX_RESTORE_TIMER = 0.1/BOX_RESTORE_TIMER = 0.5/' /var/lib/kytos/napps/kytos/flow_manager/settings.py
+ kytosd --help
+ sed -i s/WARNING/INFO/g /etc/kytos/logging.ini
+ test -z 'tests/ -k test_015_create_evc_intra_switch'
+ python -m pytest tests/ -k test_015_create_evc_intra_switch
++ date +%Y%m%d%H%M%S
+ tee log-e2e-20220114101147
============================= test session starts ==============================
platform linux2 -- Python 2.7.16, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
rootdir: /kytos-end-to-end-tests, inifile:
plugins: timeout-1.3.3
collected 189 items / 188 deselected

tests/test_e2e_10_mef_eline.py .                                         [100%]

------------------------------- start/stop times -------------------------------
================== 1 passed, 188 deselected in 83.03 seconds ===================
```

Reply:
@italovalcy, I've just tried it, the example passed, and also tried out a forced fail example, and in that case I the timestamp was there, really great to have this: 


```
============================= test session starts ==============================
platform linux2 -- Python 2.7.16, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
rootdir: /tests, inifile:
plugins: timeout-1.3.3
collected 189 items / 188 deselected

tests/test_e2e_10_mef_eline.py .                                         [100%]

------------------------------- start/stop times -------------------------------
```


```
tests/test_e2e_10_mef_eline.py F                                         [100%]

=================================== FAILURES ===================================
_______________ TestE2EMefEline.test_015_create_evc_intra_switch _______________

self = <tests.test_e2e_10_mef_eline.TestE2EMefEline instance at 0x7f6b16a277a0>

    def test_015_create_evc_intra_switch(self):
        """ create an intra-switch EVC e-line with VLAN tag
        (UNIs in the same switch) """
        payload = {
            "name": "my evc1",
            "enabled": True,
            "uni_a": {
                "interface_id": "00:00:00:00:00:00:00:01:1",
                "tag": {
                    "tag_type": 1,
                    "value": 101
                }
            },
            "uni_z": {
                "interface_id": "00:00:00:00:00:00:00:01:2",
                "tag": {
                    "tag_type": 1,
                    "value": 101
                }
            }
        }
        api_url = KYTOS_API + '/mef_eline/v2/evc/'
        response = requests.post(api_url, json=payload)
        assert response.status_code == 201
        data = response.json()
        assert 'circuit_id' in data
>       assert False
E       assert False

tests/test_e2e_10_mef_eline.py:109: AssertionError
---------------------------- Captured stderr setup -----------------------------
*** Error setting resource limits. Mininet's performance may be affected.
Unable to contact the remote controller at 127.0.0.1:6653
Terminating on signal 15
------------------------------- start/stop times -------------------------------
test_e2e_10_mef_eline.py::TestE2EMefEline::()::test_015_create_evc_intra_switch: 2022-01-
14,13:04:33.005067 - 2022-01-14,13:04:33.066025
================== 1 failed, 188 deselected in 47.16 seconds ===================
```